[AIRFLOW-947] Improve exceptions for unavailable Presto cluster

artwr opened the pull request:
This improves error logging when the Presto cluster is unavailable
and the underlying error is a 503 http response. This introspects
the error to prevent trying to access the 'message' attribute when
not present.

Please accept this PR that addresses the following issues:
- [AIRFLOW-947](https://issues.apache.org/jira/browse/AIRFLOW-947)

Testing Done:
- This would be more of an integration test. I will work on it. Please consider this WIP.

artwr committed:
[AIRFLOW-947] Improve exceptions for unavailable Presto cluster

This improves error logging when the Presto cluster is unavailable
and the underlying error is a 503 http response. This introspects
the error to prevent trying to access the 'message' attribute when
not present.

mistercrunch commented on airflow/hooks/presto_hook.py:
```diff
@@ -80,7 +80,14 @@ def get_first(self, hql, parameters=None):
                 self._strip_sql(hql), parameters)
         except DatabaseError as e:
-            raise PrestoException(e[0]['message'])
+            if (hasattr(e, 'message')
```
Would be nice the add a `self._handle_exception(e)` since the routine is copypasted

bolkedebruin committed:
Revert "[AIRFLOW-719] Prevent DAGs from ending prematurely"

This reverts commit 1fdcf2480555f06cce3fc9bba97fbf3d64f074d3.

This reinstates the previous logic (< 1.8.0) that ALL_SUCCESS requires
all tasks to be successful instead of also counting SKIPPED
tasks as part of the successful tasks.

bolkedebruin committed:
[AIRFLOW-719] Fix race condition in ShortCircuit, Branch and LatestOnly

Both the ShortCircuitOperator, Branchoperator and LatestOnlyOperator
 were arbitrarily changing the states of TaskInstances without locking
them in the database. As the scheduler checks the state of dag runs
asynchronously the dag run state could be set to failed while the
operators are updating the downstream tasks.

A better fix would to use the dag run iteself in the context of the
Operator.

bolkedebruin committed:
Merge branch 'AIRFLOW-719' into AIRFLOW-719-3

bolkedebruin committed:
Merge pull request #2195 from bolkedebruin/AIRFLOW-719

saguziel committed:
[AIRFLOW-1064] Change default sort to job_id for TaskInstanceModelView

The TaskInstanceModelView default sort column is
on an unindexed column.
We shouldn't need an index on start_date, and
job_id is just as logical
of a default sort.

Closes #2215 from saguziel/aguziel-fix-ti-page

mengxr committed:
[AIRFLOW-1067] use example.com in examples

We use airflow@airflow.com in examples. However,
https://airflow.com
is owned by a company named Airflow (selling fans,
etc). We should use
airflow@example.com instead. That domain is
created for this purpose.

Closes #2217 from mengxr/AIRFLOW-1067
